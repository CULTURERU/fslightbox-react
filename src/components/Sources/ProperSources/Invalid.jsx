import React, { Component } from 'react';
import PropTypes from "prop-types";

class Invalid extends Component {
    constructor(props) {
        super(props);
        if (!this.props.sourcesData.isSourceAlreadyLoadedArray[this.props.i]) {
            this.props.sourcesData.isSourceAlreadyLoadedArray[this.props.i] = true;
        }
    }

    render() {
        return (
            <div
                className="fslightbox-invalid-file-wrapper fslightbox-flex-centered"
                ref={ this.props.sources[this.props.i] }>
                Invalid file
            </div>
        );
    }
}

Invalid.propTypes = {
    sources: PropTypes.array.isRequired,
    sourcesData: PropTypes.object.isRequired,
    i: PropTypes.number.isRequired
};
export default Invalid;