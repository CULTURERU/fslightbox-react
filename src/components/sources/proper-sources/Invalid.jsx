import React, { useEffect } from 'react';
import { FADE_IN_STRONG_CLASS_NAME, FLEX_CENTERED_CLASS_NAME, PREFIX } from "../../../constants/classes-names";

const Invalid = (
    {
        fsLightbox: {
            componentsServices: { isSourceLoadedCollection },
            elements: { sourcesOuters }
        }, i
    }
) => {
    useEffect(() => {
        isSourceLoadedCollection[i].set(true);
        sourcesOuters[i].current.classList.add(FADE_IN_STRONG_CLASS_NAME);
    });

    return (
        <div className={ `${ PREFIX }invalid-file-wrapper ${ FLEX_CENTERED_CLASS_NAME }` }>
            Invalid source
        </div>
    );
};

export default Invalid;
