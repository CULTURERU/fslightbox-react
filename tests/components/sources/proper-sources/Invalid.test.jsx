import React from 'react';
import Invalid from "../../../../src/components/sources/proper-sources/Invalid";
import { mount, shallow } from "enzyme";

const fsLightbox = {
    core: {
        sourceAnimator: {
            animateSourceFromSlide: () => ({
                fadeIn: () => {}
            })
        }
    },
    elements: {
        sources: [{
            current: null
        }]
    }
};

let invalid;

describe('ref to sources array in fsLightbox object', () => {
    beforeAll(() => {
        invalid = mount(<Invalid fsLightbox={ fsLightbox } index={ 0 }/>);
    });

    it('should be equal to Invalid', () => {
        expect(fsLightbox.elements.sources[0].current).toEqual(invalid.getDOMNode());
    });
});

describe('on render', () => {
    let fadeIn;

    beforeAll(() => {
        fadeIn = jest.fn();
        fsLightbox.core.sourceAnimator.animateSourceFromSlide = jest.fn(() => ({
            fadeIn: fadeIn
        }));
        invalid = mount(<Invalid fsLightbox={ fsLightbox } index={ 0 }/>);
    });

    it('should call animateSourceFromSlide', () => {
        expect(fsLightbox.core.sourceAnimator.animateSourceFromSlide).toBeCalled();
    });

    it('should call fadeIn', () => {
        expect(fadeIn).toBeCalled();
    });
});

describe('Invalid DOM', () => {
    beforeAll(() => {
        invalid = shallow(<Invalid fsLightbox={ fsLightbox } index={ 0 }/>);
    });

    it('should match snapshot', () => {
        expect(invalid).toMatchSnapshot();
    });
});