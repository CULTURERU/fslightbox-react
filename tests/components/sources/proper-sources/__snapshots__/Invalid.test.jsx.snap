// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOM should match snapshot 1`] = `
<div
  className="fslightbox-invalid-file-wrapper fslightbox-flex-centered"
>
  Invalid file
</div>
`;

exports[`Invalid DOM should match snapshot 1`] = `
<div
  className="fslightbox-invalid-file-wrapper fslightbox-flex-centered"
>
  Invalid file
</div>
`;
